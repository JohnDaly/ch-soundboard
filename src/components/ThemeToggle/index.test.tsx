// External Dependencies
import { mount } from 'enzyme'
import enzymeToJSON from 'enzyme-to-json'
import 'jest-styled-components'
import React from 'react'

// Internal Dependencies
import { ThemeToggle } from './index'

describe('ThemeToggle', () => {
    it('renders without crashing', () => {
        const wrapper = mount(<ThemeToggle isDarkTheme={false} onClick={jest.fn()} />)
        expect(wrapper).toBeDefined()
    })

    it('should match the snapshot (Light Theme)', () => {
        const wrapper = mount(<ThemeToggle isDarkTheme={false} onClick={jest.fn()} />)
        expect(enzymeToJSON(wrapper)).toMatchSnapshot()
    })

    it('should match the snapshot (Dark Theme)', () => {
        const wrapper = mount(<ThemeToggle isDarkTheme={true} onClick={jest.fn()} />)
        expect(enzymeToJSON(wrapper)).toMatchSnapshot()
    })
})
