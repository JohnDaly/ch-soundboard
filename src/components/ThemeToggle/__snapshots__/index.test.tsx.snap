// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ThemeToggle should match the snapshot (Dark Theme) 1`] = `
.c0 {
  margin: 5px 5px 5px auto;
}

<ThemeToggle
  isDarkTheme={true}
  onClick={[Function]}
>
  <Component
    className="c0"
    isDarkTheme={true}
    onClick={[Function]}
  >
    <button
      className="btn btn-primary c0"
      onClick={[Function]}
    >
      <ThemeIcon
        isDarkTheme={true}
      >
        <i
          className="fas fa-sun"
        />
      </ThemeIcon>
    </button>
  </Component>
</ThemeToggle>
`;

exports[`ThemeToggle should match the snapshot (Light Theme) 1`] = `
.c0 {
  margin: 5px 5px 5px auto;
}

<ThemeToggle
  isDarkTheme={false}
  onClick={[Function]}
>
  <Component
    className="c0"
    isDarkTheme={false}
    onClick={[Function]}
  >
    <button
      className="btn btn-primary c0"
      onClick={[Function]}
    >
      <ThemeIcon
        isDarkTheme={false}
      >
        <i
          className="fas fa-moon"
        />
      </ThemeIcon>
    </button>
  </Component>
</ThemeToggle>
`;
